_docs/operations/upgrades.md

```diff
@@ -0,0 +1,220 @@
+anchor: operations
+module: operations/upgrades
+---
+
+# Upgrading StorageOS version
+
+> Before upgrading, check the 
+> [release notes]({%link _docs/reference/release_notes.md %}) to confirm
+> whether there is a safe upgrade path between versions.
+
+
+StorageOS version upgrades must be planned and executed taking into consideration
+that volumes will be inaccessible during the process. It is recommended to
+schedule a maintenance window.
+
+Currently there are two strategies to upgrade StorageOS and both maintain data
+integrity during the upgrade process. StorageOS keeps the data on the hosts
+where the StorageOS container is running. When the new version of StorageOS starts,
+the volumes from the previous version are available.
+
+1. Full stop of the cluster
+1. Rolling update
+
+> More upgrade procedures will be released that will automate the main part of
+> the process and fulfil use cases not covered currently.
+
+> The [StorageOS cli]({%link _docs/reference/cli/index.md %}) is required to perform an upgrade.
+
+## Option 1. Full stop of the cluster
+
+This option consists of downscaling all applications using StorageOS volumes to
+0, stopping all StorageOS pods, starting StorageOS with a
+new version and rescaling applications to previous size. Deployments that don't
+use StorageOS volumes remain unaffected.
+
+This option does not require moving data across nodes, therefore it is
+recommended for clusters with large data sets. However, __it implies downtime__ for
+stateful applications.
+
+Execute the following instructions:
+
+Prepare upgrade:
+
+1. Prepare an etcd backup if using external etcd.
+1. Optionally make sure all nodes have the new StorageOS image pulled, so the new
+   containers will start promptly. 
+   ```bash
+docker pull storageos/node:$NEW_VERSION
+   ```
+1. Downscale any applications using StorageOS volumes to 0.
+
+    > Any mount points will hang while StorageOS Pods are not present if the
+    > application Pods haven't been stopped.
+
+1. Put the StorageOS cluster in maintenance mode.
+
+    StorageOS implements a maintenance mode that freezes the cluster. When in
+    maintenance mode, StorageOS operations are limited. Functionalities such as
+    volume provisioning, failover of primary volumes or managing nodes are
+    disabled. For more details see the [Maintenance]<!-- todo --> page.
+1. Make sure the update strategy of StorageOS is `OnDelete`.
+    ```bash
+   $ export NAMESPACE=storageos
+   $ export DAEMONSET_NAME=storageos
+   $ kubectl -n $NAMESPACE get ds/$DAEMONSET_NAME -o {% raw %}go-template='{{.spec.updateStrategy.type}}{{"\n"}}'{% endraw %}
+   OnDelete
+    ```
+
+Execute update:
+
+1. Change the StorageOS version for the StorageOS node container
+
+    ```bash
+   kubectl -n $NAMESPACE set image ds/$DAEMONSET_NAME storageos=storageos/node:$NEW_VERSION
+    ```
+
+1. Delete StorageOS Pods.
+
+    ```bash
+   kubectl -n $NAMESPACE delete --selector app=storageos 
```

Comment:
This results in error:
```
$ kubectl -n storageos delete --selector=app=storageos
error: at least one resource must be specified to use a selector
```
Should have `delete pods`.

Reply:
Would be safer to add kind=daemonset in the selector, to avoid deleting the CSI statefulset pods.
```
kubectl -n storageos delete pods --selector=app=storageos,kind=daemonset
```